pkg/reduce/readloop/readloop_test.go

```diff
@@ -0,0 +1,227 @@
+package readloop
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/isb"
+	"github.com/numaproj/numaflow/pkg/isb/stores/simplebuffer"
+	"github.com/numaproj/numaflow/pkg/pbq"
+	"github.com/numaproj/numaflow/pkg/pbq/partition"
+	"github.com/numaproj/numaflow/pkg/pbq/store"
+	"github.com/numaproj/numaflow/pkg/watermark/generic"
+	"github.com/numaproj/numaflow/pkg/window/strategy/fixed"
+	"github.com/stretchr/testify/assert"
+)
+
+// PayloadForTest is a dummy payload for testing.
+type PayloadForTest struct {
+	Key   string
+	Value int
+}
+
+type SumReduceTest struct {
+}
+
+func (s *SumReduceTest) WhereTo(s2 string) ([]string, error) {
+	return []string{"reduce-buffer"}, nil
+}
+
+func (s *SumReduceTest) ApplyReduce(ctx context.Context, partitionID *partition.ID, messageStream <-chan *isb.ReadMessage) ([]*isb.Message, error) {
+	sum := 0
+	for msg := range messageStream {
+		var payload PayloadForTest
+		_ = json.Unmarshal(msg.Payload, &payload)
+		sum += payload.Value
+	}
+
+	payload := PayloadForTest{Key: "sum", Value: sum}
+	b, _ := json.Marshal(payload)
+	ret := &isb.Message{
+		Header: isb.Header{
+			PaneInfo: isb.PaneInfo{
+				StartTime: partitionID.Start,
+				EndTime:   partitionID.End,
+				EventTime: partitionID.End,
+			},
+			ID:  "msgID",
+			Key: "result",
+		},
+		Body: isb.Body{Payload: b},
+	}
+	return []*isb.Message{
+		ret,
+	}, nil
+}
+
+// mock store for test
+type TestStore struct {
```

Comment:
I thought we were planning to use in-memory store to do this rather than mocking the interface. Is that still the plan?

Reply:
still the plan, I had already written this test with a mock store. I have changed it to use a memory store